operator/pkg/name/name.go

```diff
@@ -77,8 +84,32 @@ var (
 	DeprecatedNames = []ComponentName{
 		InjectorComponentName,
 	}
-	allComponentNamesMap        = make(map[ComponentName]bool)
-	deprecatedComponentNamesMap = make(map[ComponentName]bool)
+	AllLegacyAddonComponentNames = []ComponentName{
+		PrometheusComponentName,
+		KialiComponentName,
+		GrafanaComponentName,
+		TracingComponentName,
+		CoreDNSComponentName,
+	}
+	allComponentNamesMap         = make(map[ComponentName]bool)
+	deprecatedComponentNamesMap  = make(map[ComponentName]bool)
+	LegacyAddonComponentNamesMap = make(map[ComponentName]bool)
+	LegacyAddonComponentPathMap  = make(map[string]string)
+
+	// ComponentNameToHelmComponentPath defines mapping from component name to helm component root path.
+	// TODO: merge this with the componentMaps defined in translateConfig
+	ComponentNameToHelmComponentPath = map[ComponentName]string{
```

Comment:
this needs to be removed. we're hard-coding components names and also mixing translate and helm concepts into the name package (helm paths have nothing to do with naming).